examples/twittermap/web/public/javascripts/sidebar/controllers.js

```diff
@@ -1,16 +1,31 @@
 angular.module("cloudberry.sidebar", ["cloudberry.common"])
-  .controller("SidebarCtrl", function($scope, $timeout, cloudberry, moduleManager, cloudberryClient, queryUtil, cloudberryConfig) {
-
+  .controller("SidebarCtrl", function($scope, $timeout, cloudberry, moduleManager, cloudberryClient, queryUtil, cloudberryConfig, $http) {
+    
     // Flag whether current result is outdated
     $scope.isHashTagOutdated = true;
     $scope.isSampleTweetsOutdated = true;
-
     // Flag whether sidebar tab is open
     $scope.isHashTagOpen = false;
-    $scope.isSampleTweetsOpen = true;
-
+    $scope.isSampleTweetsOpen = false;
     $scope.currentTab = "sampletweetTab";
-
+    var sampleTweets = [];
+    $scope.drawTweetMode = 2; //Initially set to 2 liveTweets Mode, change this variable to 1 to enable traditional draw
+    var timeRange = 3; // Set length of time interval in seconds
+    var sendQueryLoop = {}; //Store variable for window.setInterval function, keep only one setInterval function avtive at a time
+    $scope.liveTweetsLoop = {};//Store variable for window.setInterval function, stop live tweets feeding when user specified a 
+                            //time interval in time bar
+
+    var secondLiveTweetQueryTimeOut = null;
+    var timeSeriesEnd = new Date(cloudberry.parameters.timeInterval.end);// This date will be the latest date of tweets been ingested
+    var timeZoneOffset = ((new Date).getTimezoneOffset())/60;
+    timeSeriesEnd.setHours(timeSeriesEnd.getHours()-timeZoneOffset);//consider the timezone, in order to get live tweets work in any circumstance
```

Comment:
What if the user is in a time zone ahead of Greenwich Time (e.g. China)? Will this `- timeZoneOffset` still work?

Reply:
Also there are spaces around `-`

Reply:
The time-zone offset is the difference, in minutes, between UTC and local time. Note that this means that the offset is positive if the local timezone is behind UTC and negative if it is ahead.